src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java

```diff
@@ -457,11 +569,34 @@ public String getQuestionTypeChoiceOption() {
     @Override
     public List<String> validateQuestionDetails(String courseId) {
         List<String> errors = new ArrayList<>();
-        if (generateOptionsFor == FeedbackParticipantType.NONE
-                && numOfMcqChoices < Const.FeedbackQuestion.MCQ_MIN_NUM_OF_CHOICES) {
-            errors.add(Const.FeedbackQuestion.MCQ_ERROR_NOT_ENOUGH_CHOICES
-                       + Const.FeedbackQuestion.MCQ_MIN_NUM_OF_CHOICES + ".");
+        if (generateOptionsFor == FeedbackParticipantType.NONE) {
```

Comment:
Good point. :+1: Will do.

Reply:
@xpdavid I just realised that, validating this is redundant, as this situation will never occur. We can add the validation but we can't test it even if we pass `hasAssignedWeights` as `on` when `generateOptionsFor != NONE`, as the processing of weight fields and `hasAssignedWeights` are in the `generatedMcqOptions.equals(FeedbackParticipantType.NONE.toString()` block in `extractQuestionDetails` function. What should I do in this case.

```java
        if (generatedMcqOptions.equals(FeedbackParticipantType.NONE.toString())) {
            String numMcqChoicesCreatedString =
                    HttpRequestHelper.getValueFromParamMap(requestParameters,
                                                           Const.ParamsNames.FEEDBACK_QUESTION_NUMBEROFCHOICECREATED);
            Assumption.assertNotNull("Null number of choice for MCQ", numMcqChoicesCreatedString);
            int numMcqChoicesCreated = Integer.parseInt(numMcqChoicesCreatedString);

            for (int i = 0; i < numMcqChoicesCreated; i++) {
                String paramName = Const.ParamsNames.FEEDBACK_QUESTION_MCQCHOICE + "-" + i;
                String mcqChoice = HttpRequestHelper.getValueFromParamMap(requestParameters, paramName);
                if (mcqChoice != null && !mcqChoice.trim().isEmpty()) {
                    mcqChoices.add(mcqChoice);
                    numOfMcqChoices++;
                }
            }

            String hasAssignedWeightsString = HttpRequestHelper.getValueFromParamMap(
                    requestParameters, Const.ParamsNames.FEEDBACK_QUESTION_MCQ_HAS_WEIGHTS_ASSIGNED);
            boolean hasAssignedWeights = "on".equals(hasAssignedWeightsString);
            List<Double> mcqWeights = getMcqWeights(
                    requestParameters, numMcqChoicesCreated, hasAssignedWeights);
            double mcqOtherWeight = getMcqOtherWeight(requestParameters, mcqOtherEnabled, hasAssignedWeights);
            setMcqQuestionDetails(
                    numOfMcqChoices, mcqChoices, mcqOtherEnabled, hasAssignedWeights, mcqWeights, mcqOtherWeight);
        } else {
            setMcqQuestionDetails(FeedbackParticipantType.valueOf(generatedMcqOptions));
        }
```

Reply:
Instead of doing a validation for this, I wrote a unit test for `extractQuestionDetails` to test that the above mentioned case is covered in our logic.